src/metabase/api/defendpoint2.clj

```diff
@@ -0,0 +1,235 @@
+(ns metabase.api.defendpoint2
+  (:require
+   [clojure.spec.alpha :as s]
+   [clojure.string :as str]
+   [compojure.core :as compojure]
+   [malli.core :as mc]
+   [malli.transform :as mtx]
+   [metabase.api.common.internal :as internal]
+   [metabase.config :as config]
+   [metabase.util :as u]
+   [metabase.util.log :as log]
+   [metabase.util.malli.schema :as ms]
+   [ring.middleware.multipart-params :as mp]
+   [metabase.util.malli.describe :as umd]
+   [malli.error :as me]
+   [metabase.util.malli :as mu]))
+
+;;; almost a copy of malli.experimental.lite, but optional is defined by [:maybe]
+
+(declare make-schema)
+
+(def ^{:dynamic true :private true} *options* nil)
+(defn- -entry [[k v]]
+  (let [optional (and (vector? v)
+                      (= :maybe (first v)))]
+    (cond-> [k] optional (conj {:optional true}) :always (conj (make-schema v)))))
+
+(defn make-schema
+  "Compile map-based syntax to Malli schema."
+  [x]
+  (mc/schema (if (map? x)
+               (into [:map {:closed config/is-dev?}] (map -entry x))
```

Comment:
I love this, but I'd be way better if we can make this log/warn missing keys during development, that way we have signal to update the schema.

Reply:
Well.. it's using `log/warn` right now. :) I think just failing requests in development is a much stronger signal. 😁